modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1010,6 +1079,30 @@ public List<String> assignments() {
                 .collect(Collectors.toList());
     }
 
+    /** {@inheritDoc} */
+    // TODO: IGNITE-17256 Use a placement driver for getting a primary replica.
+    @Override
+    public List<IgniteBiTuple<String, Long>> leaderAssignmentsWithTerm() {
+        List<Entry<RaftGroupService>> entries = new ArrayList<>(partitionMap.int2ObjectEntrySet());
+        List<CompletableFuture<LeaderWithTerm>> futs = new ArrayList<>();
+
+        entries.sort(Comparator.comparingInt(Entry::getIntKey));
+
+        for (Entry<RaftGroupService> e : entries) {
+            futs.add(e.getValue().refreshAndGetLeaderWithTerm());
+        }
+
+        List<IgniteBiTuple<String, Long>> leadersWithTerm = new ArrayList<>(entries.size());
+
+        for (CompletableFuture<LeaderWithTerm> fut : futs) {
+            LeaderWithTerm leaderWithTerm = fut.join();
+
+            leadersWithTerm.add(new IgniteBiTuple<>(leaderWithTerm.leader().consistentId(), leaderWithTerm.term()));
```

Comment:
you can use NodeWithTerm instantly instead of BiTuple

Reply:
AFAIK, I can't do it without changing dependencies.

`LeaderWithTerm` is located in raft-api module
`NodeWithTerm` is located in sql-engine module (it depends on table module, but not raft-api)
`InternalTableImpl` is located in table module (it depends on raft-api, but not on sql-engine)